numba/cpython/tupleobj.py

```diff
@@ -341,6 +342,12 @@ def static_getitem_tuple(context, builder, sig, args):
     elif isinstance(idx, slice):
         items = cgutils.unpack_tuple(builder, tup)[idx]
         res = context.make_tuple(builder, sig.return_type, items)
+    elif isinstance(tupty, types.LiteralStrKeyDict):
+        # pretend to be a dictionary
+        if isinstance(idxty, types.Literal):
```

Comment:
Looks like the `else` branch is not handled.

Reply:
will remove, `LiteralStrKeyDict` inherits from `Literal` so it is unreachable, think it's left over from trying to coerce a LiteralDict into a TypedDict based on type inference spotting mutation.